pommerman/network/ion_client/__init__.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+"""IonClient Entry Point  
+
+This library allows playing of matches on network via the WebSockets protocol.  
+Functions:  
+init() - If you want to run the application normally  
+intent(network) - If you want to pass a pre-existing network object but want to
+prompt the user about what they want to do  
+match(network, room=False, import_name=False, agent_class=False) - If you want 
+to start a match directly  
+replay(network, id=False) - If you want to start a replay directly  
+"""
+
+import ui
+from . import constants
+# This is for switching between languages incase there are more
+Strings = constants.LangEN
+from .network import Network
+import signal
+
+
+def _exit_handler(_s=None, _h=None):
+    """Arguments:  
+    * _s: Unused argument  
+    * _h: Unused argument  
+    Description: Handle exiting the application"""
+    ui.info(ui.yellow, "Exiting..")
+    exit(0)
+
+
+def init():
+    """Description: Starts up the application normally by asking the user about
+    the server they want to connect to"""
+    signal.signal(signal.SIGINT, _exit_handler)
+    if ui.ask_yes_no(Strings.server_prompt.value):
+        domain = ui.ask_string(Strings.server_ip.value)
+        if domain is None:
+            ui.fatal(Strings.error_invalid_domain.value)
+    else:
+        domain = "play.pommerman.com:5050"
+    ui.info(
+        Strings.server_connecting_p1.value, ui.yellow,
+        Strings.server_connecting_p2.value, ui.reset,
+        Strings.server_connecting_p3.value
+    )
+    network = Network(domain)
+    try:
+        status = network.check()
+    except Exception as e:
+        ui.fatal(e)
```

Comment:
what is this for? i dont think i understand why you are catching the exception just to throw a fatal? does ui talk to other threads or ...?

Reply:
`ion_client` only has one thread. The exceptions are a part of a partial rewrite of the client side Network class, if they fail they `raise` an exception which is a string denoting the error.

Reply:
This is the UI library that I use. It'll give you more context: https://supertanker.github.io/python-cli-ui/

Reply:
This doesn't quite answer my question. You could just let the exception be thrown and then the user would see where the error happened and what the error is. Instead, you catch the exception and then throw another one ... with the original Exception as the one you threw. This seems redundant. Why is it necessary?

Reply:
The UI library's fatal prints the exception in the format "Error: The exception is a string so it'll be here" instead of producing a traceback which isn't required for these sorts of exceptions.

Reply:
Have you ever worked with a system that hid the traceback? It's ... hard (infuriating?) to track down bugs.